src/yggdrasil/api.go

```diff
@@ -225,32 +226,47 @@ func (c *Core) GetSessions() []Session {
 	return sessions
 }
 
-// ConnListen returns a listener for Yggdrasil session connections. You can only
-// call this function once as each Yggdrasil node can only have a single
-// ConnListener. Make sure to keep the reference to this for as long as it is
-// needed.
-func (c *Core) ConnListen() (*Listener, error) {
-	c.router.sessions.listenerMutex.Lock()
-	defer c.router.sessions.listenerMutex.Unlock()
-	if c.router.sessions.listener != nil {
-		return nil, errors.New("a listener already exists")
-	}
-	c.router.sessions.listener = &Listener{
-		core:  c,
-		conn:  make(chan *Conn),
-		close: make(chan interface{}),
-	}
-	return c.router.sessions.listener, nil
-}
-
-// ConnDialer returns a dialer for Yggdrasil session connections. Since
-// ConnDialers are stateless, you can request as many dialers as you like,
-// although ideally you should request only one and keep the reference to it for
-// as long as it is needed.
-func (c *Core) ConnDialer() (*Dialer, error) {
-	return &Dialer{
-		core: c,
-	}, nil
+// PacketConn returns a net.PacketConn which can be used to send and receive
+// information over Yggdrasil sessions.
+func (c *Core) PacketConn() *PacketConn {
```

Comment:
So I think ideally we should make the `Core` export the `PacketConn` interface directly, to make it clear that there'e exactly 1 `PacketConn` available to the `Core`. As-is, people could be confused that multiple calls return a pointer to the same underlying `PacketConn` (which may break things outright if they try to use the `PacketConn` in existing code -- multiple things trying to read from the same underlying `PacketConn` could make them all think there's horrible packet loss, or they could call `Close` or the various `Deadline` functions and mess things up for eachother). At the very least, we should make it clear in the documentation that it's a pointer to the same `PacketConn`.

Reply:
Agreed on making `Core` implement `PacketConn`.